build_stats.sh

```diff
@@ -0,0 +1,40 @@
+#! /bin/sh
+
+set -e
+
+HERE_DIR="$(cd "$(dirname "$1")"; pwd)/$(basename "$1")"
```

Comment:
OK, so this is one of those things where to be fully portable you have to dance with both feet up in the air whilst simultaneously having no other part of your body touching the ground.

But a pragmatic approach is simply:
```
HERE_DIR=`dirname $_`
```
For example if I have a file `/tmp/woo.sh` which prints out using the "cd" mechanism and then the "$_" mechanism I get:
```sh
$ cd /home/ltratt
$ PATH=/tmp/ woo.sh
/home/ltratt
/tmp/
```
i.e. because the "cd" mechanism uses `pwd` it picks up where I'm executing the command from, even though that has nothing to do with the script.